### ISSUE TITLE
Can't build release after upgrading plugin

### ISSUE THREAD

**Issue by yzargari** [phonegap/phonegap-plugin-push#556]
Hi guys,

I recently upgraded to 1.5.3 and since then, I can't build with the release switch. It builds fine without it. Any idea why?

Below is the output of running  "phonegap build ios --verbose --device --release".

Thanks,
Joseph.

```
[36m[phonegap] [39m executing 'cordova build ios --verbose --device --release'...
Executing "before_build"  hook for all plugins.

Executing "before_prepare"  hook for all plugins.

Searching PlatformJson files for differences between project vs. platform installed plugins

No differences found between project and ios platform. Continuing...

Generating config.xml from defaults for platform "ios"

Wrote out iOS Bundle Identifier to "com.myappname.app"

Wrote out iOS Bundle Version to "1.0.5"

Copying icon from /Users/admin/Desktop/myapppdir/www/res/icon/ios/icon-60.png to /Users/admin/Desktop/myapppdir/platforms/ios/MyAppName/Resources/icons/icon-60.png

Copying icon from /Users/admin/Desktop/myapppdir/www/res/icon/ios/icon-60-2x.png to /Users/admin/Desktop/myapppdir/platforms/ios/MyAppName/Resources/icons/icon-60@2x.png

Copying icon from /Users/admin/Desktop/myapppdir/www/res/icon/ios/icon-60-3x.png to /Users/admin/Desktop/myapppdir/platforms/ios/MyAppName/Resources/icons/icon-60@3x.png

Copying icon from /Users/admin/Desktop/myapppdir/www/res/icon/ios/icon-76.png to /Users/admin/Desktop/myapppdir/platforms/ios/MyAppName/Resources/icons/icon-76.png

Copying icon from /Users/admin/Desktop/myapppdir/www/res/icon/ios/icon-76-2x.png to /Users/admin/Desktop/myapppdir/platforms/ios/MyAppName/Resources/icons/icon-76@2x.png

Copying icon from /Users/admin/Desktop/myapppdir/www/res/icon/ios/icon-40.png to /Users/admin/Desktop/myapppdir/platforms/ios/MyAppName/Resources/icons/icon-40.png

Copying icon from /Users/admin/Desktop/myapppdir/www/res/icon/ios/icon-40-2x.png to /Users/admin/Desktop/myapppdir/platforms/ios/MyAppName/Resources/icons/icon-40@2x.png

Copying icon from /Users/admin/Desktop/myapppdir/www/res/icon/ios/icon-57.png to /Users/admin/Desktop/myapppdir/platforms/ios/MyAppName/Resources/icons/icon.png

Copying icon from /Users/admin/Desktop/myapppdir/www/res/icon/ios/icon-57-2x.png to /Users/admin/Desktop/myapppdir/platforms/ios/MyAppName/Resources/icons/icon@2x.png

Copying icon from /Users/admin/Desktop/myapppdir/www/res/icon/ios/icon-72.png to /Users/admin/Desktop/myapppdir/platforms/ios/MyAppName/Resources/icons/icon-72.png

Copying icon from /Users/admin/Desktop/myapppdir/www/res/icon/ios/icon-72-2x.png to /Users/admin/Desktop/myapppdir/platforms/ios/MyAppName/Resources/icons/icon-72@2x.png

Copying splash from /Users/admin/Desktop/myapppdir/www/res/screen/ios/Default~iphone.png to /Users/admin/Desktop/myapppdir/platforms/ios/MyAppName/Resources/splash/Default~iphone.png

Copying splash from /Users/admin/Desktop/myapppdir/www/res/screen/ios/Default@2x~iphone.png to /Users/admin/Desktop/myapppdir/platforms/ios/MyAppName/Resources/splash/Default@2x~iphone.png

Copying splash from /Users/admin/Desktop/myapppdir/www/res/screen/ios/Default-Portrait~ipad.png to /Users/admin/Desktop/myapppdir/platforms/ios/MyAppName/Resources/splash/Default-Portrait~ipad.png

Copying splash from /Users/admin/Desktop/myapppdir/www/res/screen/ios/Default-Portrait@2x~ipad.png to /Users/admin/Desktop/myapppdir/platforms/ios/MyAppName/Resources/splash/Default-Portrait@2x~ipad.png

Copying splash from /Users/admin/Desktop/myapppdir/www/res/screen/ios/Default-568h@2x~iphone.png to /Users/admin/Desktop/myapppdir/platforms/ios/MyAppName/Resources/splash/Default-568h@2x~iphone.png

Copying splash from /Users/admin/Desktop/myapppdir/www/res/screen/ios/Default-667h.png to /Users/admin/Desktop/myapppdir/platforms/ios/MyAppName/Resources/splash/Default-667h.png

Copying splash from /Users/admin/Desktop/myapppdir/www/res/screen/ios/Default-736h.png to /Users/admin/Desktop/myapppdir/platforms/ios/MyAppName/Resources/splash/Default-736h.png

Set TARGETED_DEVICE_FAMILY to "1,2".

iOS Product Name has not changed (still "MyAppName")

Executing "after_prepare"  hook for all plugins.

Executing "before_compile"  hook for all plugins.

Running command: /Users/admin/Desktop/myapppdir/platforms/ios/cordova/build --release --device

Building project  : /Users/admin/Desktop/myapppdir/platforms/ios/MyAppName.xcodeproj

	Configuration : Release

	Platform      : device

Build settings from command line:

    ARCHS = armv7 armv7s arm64
    CONFIGURATION_BUILD_DIR = /Users/admin/Desktop/myapppdir/platforms/ios/build/device
    SDKROOT = iphoneos9.2
    SHARED_PRECOMPS_DIR = /Users/admin/Desktop/myapppdir/platforms/ios/build/sharedpch
    VALID_ARCHS = armv7 armv7s arm64

Build settings from configuration file '/Users/admin/Desktop/myapppdir/platforms/ios/cordova/build-release.xcconfig':
    CODE_SIGN_IDENTITY = iPhone Distribution
    ENABLE_BITCODE = NO


=== BUILD TARGET CordovaLib OF PROJECT CordovaLib WITH CONFIGURATION Release ===




Check dependencies



=== BUILD TARGET MyAppName OF PROJECT MyAppName WITH CONFIGURATION Release ===


Check dependencies



PhaseScriptExecution Copy\ www\ directory build/MyAppName.build/Release-iphoneos/MyAppName.build/Script-304B58A110DAC018002A0835.sh
    cd /Users/admin/Desktop/myapppdir/platforms/ios
    /bin/sh -c /Users/admin/Desktop/myapppdir/platforms/ios/build/MyAppName.build/Release-iphoneos/MyAppName.build/Script-304B58A110DAC018002A0835.sh



CopyPNGFile build/device/MyAppName.app/icon-76.png MyAppName/Resources/icons/icon-76.png
    cd /Users/admin/Desktop/myapppdir/platforms/ios
    export PATH="/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/usr/bin:/Applications/Xcode.app/Contents/Developer/usr/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin"
    /Applications/Xcode.app/Contents/Developer/usr/bin/copypng -compress /Users/admin/Desktop/myapppdir/platforms/ios/MyAppName/Resources/icons/icon-76.png /Users/admin/Desktop/myapppdir/platforms/ios/build/device/MyAppName.app/icon-76.png



CopyPNGFile build/device/MyAppName.app/icon-40@2x.png MyAppName/Resources/icons/icon-40@2x.png
    cd /Users/admin/Desktop/myapppdir/platforms/ios
    export PATH="/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/usr/bin:/Applications/Xcode.app/Contents/Developer/usr/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin"
    /Applications/Xcode.app/Contents/Developer/usr/bin/copypng -compress /Users/admin/Desktop/myapppdir/platforms/ios/MyAppName/Resources/icons/icon-40@2x.png /Users/admin/Desktop/myapppdir/platforms/ios/build/device/MyAppName.app/icon-40@2x.png


CopyPNGFile build/device/MyAppName.app/icon-72.png MyAppName/Resources/icons/icon-72.png
    cd /Users/admin/Desktop/myapppdir/platforms/ios
    export PATH="/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/usr/bin:/Applications/Xcode.app/Contents/Developer/usr/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin"
    /Applications/Xcode.app/Contents/Developer/usr/bin/copypng -compress /Users/admin/Desktop/myapppdir/platforms/ios/MyAppName/Resources/icons/icon-72.png /Users/admin/Desktop/myapppdir/platforms/ios/build/device/MyAppName.app/icon-72.png

CopyPNGFile build/device/MyAppName.app/icon.png MyAppName/Resources/icons/icon.png
    cd /Users/admin/Desktop/myapppdir/platforms/ios
    export PATH="/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/usr/bin:/Applications/Xcode.app/Contents/Developer/usr/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin"
    /Applications/Xcode.app/Contents/Developer/usr/bin/copypng -compress /Users/admin/Desktop/myapppdir/platforms/ios/MyAppName/Resources/icons/icon.png /Users/admin/Desktop/myapppdir/platforms/ios/build/device/MyAppName.app/icon.png

CopyPNGFile build/device/MyAppName.app/Default-736h.png MyAppName/Resources/splash/Default-736h.png
    cd /Users/admin/Desktop/myapppdir/platforms/ios
    export PATH="/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/usr/bin:/Applications/Xcode.app/Contents/Developer/usr/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin"
    /Applications/Xcode.app/Contents/Developer/usr/bin/copypng -compress /Users/admin/Desktop/myapppdir/platforms/ios/MyAppName/Resources/splash/Default-736h.png /Users/admin/Desktop/myapppdir/platforms/ios/build/device/MyAppName.app/Default-736h.png


CopyPNGFile build/device/MyAppName.app/icon@2x.png MyAppName/Resources/icons/icon@2x.png
    cd /Users/admin/Desktop/myapppdir/platforms/ios
    export PATH="/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/usr/bin:/Applications/Xcode.app/Contents/Developer/usr/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin"
    /Applications/Xcode.app/Contents/Developer/usr/bin/copypng -compress /Users/admin/Desktop/myapppdir/platforms/ios/MyAppName/Resources/icons/icon@2x.png /Users/admin/Desktop/myapppdir/platforms/ios/build/device/MyAppName.app/icon@2x.png


CopyPNGFile build/device/MyAppName.app/icon-60@2x.png MyAppName/Resources/icons/icon-60@2x.png
    cd /Users/admin/Desktop/myapppdir/platforms/ios
    export PATH="/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/usr/bin:/Applications/Xcode.app/Contents/Developer/usr/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin"
    /Applications/Xcode.app/Contents/Developer/usr/bin/copypng -compress /Users/admin/Desktop/myapppdir/platforms/ios/MyAppName/Resources/icons/icon-60@2x.png /Users/admin/Desktop/myapppdir/platforms/ios/build/device/MyAppName.app/icon-60@2x.png

CopyPNGFile build/device/MyAppName.app/Default-Portrait@2x~ipad.png MyAppName/Resources/splash/Default-Portrait@2x~ipad.png
    cd /Users/admin/Desktop/myapppdir/platforms/ios
    export PATH="/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/usr/bin:/Applications/Xcode.app/Contents/Developer/usr/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin"
    /Applications/Xcode.app/Contents/Developer/usr/bin/copypng -compress /Users/admin/Desktop/myapppdir/platforms/ios/MyAppName/Resources/splash/Default-Portrait@2x~ipad.png /Users/admin/Desktop/myapppdir/platforms/ios/build/device/MyAppName.app/Default-Portrait@2x~ipad.png


CopyPNGFile build/device/MyAppName.app/Default-Portrait~ipad.png MyAppName/Resources/splash/Default-Portrait~ipad.png
    cd /Users/admin/Desktop/myapppdir/platforms/ios
    export PATH="/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/usr/bin:/Applications/Xcode.app/Contents/Developer/usr/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin"
    /Applications/Xcode.app/Contents/Developer/usr/bin/copypng -compress /Users/admin/Desktop/myapppdir/platforms/ios/MyAppName/Resources/splash/Default-Portrait~ipad.png /Users/admin/Desktop/myapppdir/platforms/ios/build/device/MyAppName.app/Default-Portrait~ipad.png


CopyPNGFile build/device/MyAppName.app/Default-667h.png MyAppName/Resources/splash/Default-667h.png
    cd /Users/admin/Desktop/myapppdir/platforms/ios
    export PATH="/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/usr/bin:/Applications/Xcode.app/Contents/Developer/usr/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin"
    /Applications/Xcode.app/Contents/Developer/usr/bin/copypng -compress /Users/admin/Desktop/myapppdir/platforms/ios/MyAppName/Resources/splash/Default-667h.png /Users/admin/Desktop/myapppdir/platforms/ios/build/device/MyAppName.app/Default-667h.png

CopyPNGFile build/device/MyAppName.app/icon-40.png MyAppName/Resources/icons/icon-40.png
    cd /Users/admin/Desktop/myapppdir/platforms/ios
    export PATH="/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/usr/bin:/Applications/Xcode.app/Contents/Developer/usr/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin"
    /Applications/Xcode.app/Contents/Developer/usr/bin/copypng -compress /Users/admin/Desktop/myapppdir/platforms/ios/MyAppName/Resources/icons/icon-40.png /Users/admin/Desktop/myapppdir/platforms/ios/build/device/MyAppName.app/icon-40.png

CopyPNGFile build/device/MyAppName.app/Default@2x~iphone.png MyAppName/Resources/splash/Default@2x~iphone.png
    cd /Users/admin/Desktop/myapppdir/platforms/ios
    export PATH="/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/usr/bin:/Applications/Xcode.app/Contents/Developer/usr/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin"
    /Applications/Xcode.app/Contents/Developer/usr/bin/copypng -compress /Users/admin/Desktop/myapppdir/platforms/ios/MyAppName/Resources/splash/Default@2x~iphone.png /Users/admin/Desktop/myapppdir/platforms/ios/build/device/MyAppName.app/Default@2x~iphone.png

CopyPNGFile build/device/MyAppName.app/icon-60.png MyAppName/Resources/icons/icon-60.png
    cd /Users/admin/Desktop/myapppdir/platforms/ios
    export PATH="/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/usr/bin:/Applications/Xcode.app/Contents/Developer/usr/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin"
    /Applications/Xcode.app/Contents/Developer/usr/bin/copypng -compress /Users/admin/Desktop/myapppdir/platforms/ios/MyAppName/Resources/icons/icon-60.png /Users/admin/Desktop/myapppdir/platforms/ios/build/device/MyAppName.app/icon-60.png

CopyPNGFile build/device/MyAppName.app/Default~iphone.png MyAppName/Resources/splash/Default~iphone.png
    cd /Users/admin/Desktop/myapppdir/platforms/ios
    export PATH="/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/usr/bin:/Applications/Xcode.app/Contents/Developer/usr/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin"
    /Applications/Xcode.app/Contents/Developer/usr/bin/copypng -compress /Users/admin/Desktop/myapppdir/platforms/ios/MyAppName/Resources/splash/Default~iphone.png /Users/admin/Desktop/myapppdir/platforms/ios/build/device/MyAppName.app/Default~iphone.png

CopyPNGFile build/device/MyAppName.app/Default-568h@2x~iphone.png MyAppName/Resources/splash/Default-568h@2x~iphone.png
    cd /Users/admin/Desktop/myapppdir/platforms/ios
    export PATH="/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/usr/bin:/Applications/Xcode.app/Contents/Developer/usr/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin"
    /Applications/Xcode.app/Contents/Developer/usr/bin/copypng -compress /Users/admin/Desktop/myapppdir/platforms/ios/MyAppName/Resources/splash/Default-568h@2x~iphone.png /Users/admin/Desktop/myapppdir/platforms/ios/build/device/MyAppName.app/Default-568h@2x~iphone.png

CopyPNGFile build/device/MyAppName.app/icon-60@3x.png MyAppName/Resources/icons/icon-60@3x.png
    cd /Users/admin/Desktop/myapppdir/platforms/ios
    export PATH="/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/usr/bin:/Applications/Xcode.app/Contents/Developer/usr/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin"
    /Applications/Xcode.app/Contents/Developer/usr/bin/copypng -compress /Users/admin/Desktop/myapppdir/platforms/ios/MyAppName/Resources/icons/icon-60@3x.png /Users/admin/Desktop/myapppdir/platforms/ios/build/device/MyAppName.app/icon-60@3x.png

CopyPNGFile build/device/MyAppName.app/icon-76@2x.png MyAppName/Resources/icons/icon-76@2x.png
    cd /Users/admin/Desktop/myapppdir/platforms/ios
    export PATH="/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/usr/bin:/Applications/Xcode.app/Contents/Developer/usr/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin"
    /Applications/Xcode.app/Contents/Developer/usr/bin/copypng -compress /Users/admin/Desktop/myapppdir/platforms/ios/MyAppName/Resources/icons/icon-76@2x.png /Users/admin/Desktop/myapppdir/platforms/ios/build/device/MyAppName.app/icon-76@2x.png

CopyPNGFile build/device/MyAppName.app/icon-72@2x.png MyAppName/Resources/icons/icon-72@2x.png
    cd /Users/admin/Desktop/myapppdir/platforms/ios
    export PATH="/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/usr/bin:/Applications/Xcode.app/Contents/Developer/usr/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin"
    /Applications/Xcode.app/Contents/Developer/usr/bin/copypng -compress /Users/admin/Desktop/myapppdir/platforms/ios/MyAppName/Resources/icons/icon-72@2x.png /Users/admin/Desktop/myapppdir/platforms/ios/build/device/MyAppName.app/icon-72@2x.png


ProcessInfoPlistFile build/device/MyAppName.app/Info.plist MyAppName/MyAppName-Info.plist
    cd /Users/admin/Desktop/myapppdir/platforms/ios
    export PATH="/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/usr/bin:/Applications/Xcode.app/Contents/Developer/usr/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin"
    builtin-infoPlistUtility /Users/admin/Desktop/myapppdir/platforms/ios/MyAppName/MyAppName-Info.plist -genpkginfo /Users/admin/Desktop/myapppdir/platforms/ios/build/device/MyAppName.app/PkgInfo -expandbuildsettings -format binary -platform iphoneos -additionalcontentfile /Users/admin/Desktop/myapppdir/platforms/ios/build/MyAppName.build/Release-iphoneos/MyAppName.build/MainViewController-PartialInfo.plist -additionalcontentfile /Users/admin/Desktop/myapppdir/platforms/ios/build/MyAppName.build/Release-iphoneos/MyAppName.build/scannerOverlay-PartialInfo.plist -o /Users/admin/Desktop/myapppdir/platforms/ios/build/device/MyAppName.app/Info.plist



Ld build/MyAppName.build/Release-iphoneos/MyAppName.build/Objects-normal/armv7s/MyAppName normal armv7s
    cd /Users/admin/Desktop/myapppdir/platforms/ios
    export IPHONEOS_DEPLOYMENT_TARGET=6.0
    export PATH="/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/usr/bin:/Applications/Xcode.app/Contents/Developer/usr/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin"
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang++ -arch armv7s -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS9.2.sdk -L/Users/admin/Desktop/myapppdir/platforms/ios/build/device -L/Users/admin/Desktop/myapppdir/platforms/ios/MyAppName/Plugins/phonegap-plugin-push/GCM/Libraries -F/Users/admin/Desktop/myapppdir/platforms/ios/build/device -filelist /Users/admin/Desktop/myapppdir/platforms/ios/build/MyAppName.build/Release-iphoneos/MyAppName.build/Objects-normal/armv7s/MyAppName.LinkFileList -miphoneos-version-min=6.0 -dead_strip -weak_framework CoreFoundation -weak_framework UIKit -weak_framework AVFoundation -weak_framework CoreMedia -weak-lSystem -ObjC -fobjc-arc -fobjc-link-runtime -framework AssetsLibrary /Users/admin/Desktop/myapppdir/platforms/ios/build/device/libCordova.a -framework CoreGraphics -framework MobileCoreServices -weak_framework ImageIO -framework CoreLocation -framework AVFoundation -weak_framework AudioToolbox -weak_framework SystemConfiguration -framework CoreTelephony -liconv -framework CoreVideo -framework QuartzCore -framework CoreImage -lGGLCloudMessaging -lGGLCore -lGGLInstanceIDLib -lGIP_Reachability -lGSDK_Overload -lGTMSessionFetcher_core -lGTMSessionFetcher_full -lGTMStackTrace -lGTM_AddressBook -lGTM_DebugUtils -lGTM_GTMURLBuilder -lGTM_KVO -lGTM_NSData+zlib -lGTM_NSDictionary+URLArguments -lGTM_NSScannerJSON -lGTM_NSStringHTML -lGTM_NSStringXML -lGTM_Regex -lGTM_RoundedRectPath -lGTM_StringEncoding -lGTM_SystemVersion -lGTM_UIFont+LineHeight -lGTM_core -lGTM_iPhone -lGcmLib -lProtocolBuffers -framework AddressBook -lsqlite3 -lz -Xlinker -dependency_info -Xlinker /Users/admin/Desktop/myapppdir/platforms/ios/build/MyAppName.build/Release-iphoneos/MyAppName.build/Objects-normal/armv7s/MyAppName_dependency_info.dat -o /Users/admin/Desktop/myapppdir/platforms/ios/build/MyAppName.build/Release-iphoneos/MyAppName.build/Objects-normal/armv7s/MyAppName

ld: warning: ignoring file /Users/admin/Desktop/myapppdir/platforms/ios/build/device/libCordova.a, file was built for archive which is not the architecture being linked (armv7s): /Users/admin/Desktop/myapppdir/platforms/ios/build/device/libCordova.a
ld: file is universal (4 slices) but does not contain a(n) armv7s slice: /Users/admin/Desktop/myapppdir/platforms/ios/MyAppName/Plugins/phonegap-plugin-push/GCM/Libraries/libGGLCloudMessaging.a file '/Users/admin/Desktop/myapppdir/platforms/ios/MyAppName/Plugins/phonegap-plugin-push/GCM/Libraries/libGGLCloudMessaging.a' for architecture armv7s
clang: error: linker command failed with exit code 1 (use -v to see invocation)



Ld build/MyAppName.build/Release-iphoneos/MyAppName.build/Objects-normal/arm64/MyAppName normal arm64
    cd /Users/admin/Desktop/myapppdir/platforms/ios
    export IPHONEOS_DEPLOYMENT_TARGET=6.0
    export PATH="/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/usr/bin:/Applications/Xcode.app/Contents/Developer/usr/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin"
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang++ -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS9.2.sdk -L/Users/admin/Desktop/myapppdir/platforms/ios/build/device -L/Users/admin/Desktop/myapppdir/platforms/ios/MyAppName/Plugins/phonegap-plugin-push/GCM/Libraries -F/Users/admin/Desktop/myapppdir/platforms/ios/build/device -filelist /Users/admin/Desktop/myapppdir/platforms/ios/build/MyAppName.build/Release-iphoneos/MyAppName.build/Objects-normal/arm64/MyAppName.LinkFileList -miphoneos-version-min=6.0 -dead_strip -weak_framework CoreFoundation -weak_framework UIKit -weak_framework AVFoundation -weak_framework CoreMedia -weak-lSystem -ObjC -fobjc-arc -fobjc-link-runtime -framework AssetsLibrary /Users/admin/Desktop/myapppdir/platforms/ios/build/device/libCordova.a -framework CoreGraphics -framework MobileCoreServices -weak_framework ImageIO -framework CoreLocation -framework AVFoundation -weak_framework AudioToolbox -weak_framework SystemConfiguration -framework CoreTelephony -liconv -framework CoreVideo -framework QuartzCore -framework CoreImage -lGGLCloudMessaging -lGGLCore -lGGLInstanceIDLib -lGIP_Reachability -lGSDK_Overload -lGTMSessionFetcher_core -lGTMSessionFetcher_full -lGTMStackTrace -lGTM_AddressBook -lGTM_DebugUtils -lGTM_GTMURLBuilder -lGTM_KVO -lGTM_NSData+zlib -lGTM_NSDictionary+URLArguments -lGTM_NSScannerJSON -lGTM_NSStringHTML -lGTM_NSStringXML -lGTM_Regex -lGTM_RoundedRectPath -lGTM_StringEncoding -lGTM_SystemVersion -lGTM_UIFont+LineHeight -lGTM_core -lGTM_iPhone -lGcmLib -lProtocolBuffers -framework AddressBook -lsqlite3 -lz -Xlinker -dependency_info -Xlinker /Users/admin/Desktop/myapppdir/platforms/ios/build/MyAppName.build/Release-iphoneos/MyAppName.build/Objects-normal/arm64/MyAppName_dependency_info.dat -o /Users/admin/Desktop/myapppdir/platforms/ios/build/MyAppName.build/Release-iphoneos/MyAppName.build/Objects-normal/arm64/MyAppName

ld: warning: ignoring file /Users/admin/Desktop/myapppdir/platforms/ios/build/device/libCordova.a, file was built for archive which is not the architecture being linked (arm64): /Users/admin/Desktop/myapppdir/platforms/ios/build/device/libCordova.a


Ld build/MyAppName.build/Release-iphoneos/MyAppName.build/Objects-normal/armv7/MyAppName normal armv7
    cd /Users/admin/Desktop/myapppdir/platforms/ios
    export IPHONEOS_DEPLOYMENT_TARGET=6.0
    export PATH="/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/usr/bin:/Applications/Xcode.app/Contents/Developer/usr/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin"
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang++ -arch armv7 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS9.2.sdk -L/Users/admin/Desktop/myapppdir/platforms/ios/build/device -L/Users/admin/Desktop/myapppdir/platforms/ios/MyAppName/Plugins/phonegap-plugin-push/GCM/Libraries -F/Users/admin/Desktop/myapppdir/platforms/ios/build/device -filelist /Users/admin/Desktop/myapppdir/platforms/ios/build/MyAppName.build/Release-iphoneos/MyAppName.build/Objects-normal/armv7/MyAppName.LinkFileList -miphoneos-version-min=6.0 -dead_strip -weak_framework CoreFoundation -weak_framework UIKit -weak_framework AVFoundation -weak_framework CoreMedia -weak-lSystem -ObjC -fobjc-arc -fobjc-link-runtime -framework AssetsLibrary /Users/admin/Desktop/myapppdir/platforms/ios/build/device/libCordova.a -framework CoreGraphics -framework MobileCoreServices -weak_framework ImageIO -framework CoreLocation -framework AVFoundation -weak_framework AudioToolbox -weak_framework SystemConfiguration -framework CoreTelephony -liconv -framework CoreVideo -framework QuartzCore -framework CoreImage -lGGLCloudMessaging -lGGLCore -lGGLInstanceIDLib -lGIP_Reachability -lGSDK_Overload -lGTMSessionFetcher_core -lGTMSessionFetcher_full -lGTMStackTrace -lGTM_AddressBook -lGTM_DebugUtils -lGTM_GTMURLBuilder -lGTM_KVO -lGTM_NSData+zlib -lGTM_NSDictionary+URLArguments -lGTM_NSScannerJSON -lGTM_NSStringHTML -lGTM_NSStringXML -lGTM_Regex -lGTM_RoundedRectPath -lGTM_StringEncoding -lGTM_SystemVersion -lGTM_UIFont+LineHeight -lGTM_core -lGTM_iPhone -lGcmLib -lProtocolBuffers -framework AddressBook -lsqlite3 -lz -Xlinker -dependency_info -Xlinker /Users/admin/Desktop/myapppdir/platforms/ios/build/MyAppName.build/Release-iphoneos/MyAppName.build/Objects-normal/armv7/MyAppName_dependency_info.dat -o /Users/admin/Desktop/myapppdir/platforms/ios/build/MyAppName.build/Release-iphoneos/MyAppName.build/Objects-normal/armv7/MyAppName



** BUILD FAILED **


The following build commands failed:
	Ld build/MyAppName.build/Release-iphoneos/MyAppName.build/Objects-normal/armv7s/MyAppName normal armv7s

(1 failure)

Error code 65 for command: xcodebuild with args: -xcconfig,/Users/admin/Desktop/myapppdir/platforms/ios/cordova/build-release.xcconfig,-project,MyAppName.xcodeproj,ARCHS=armv7 armv7s arm64,-target,MyAppName,-configuration,Release,-sdk,iphoneos,build,VALID_ARCHS=armv7 armv7s arm64,CONFIGURATION_BUILD_DIR=/Users/admin/Desktop/myapppdir/platforms/ios/build/device,SHARED_PRECOMPS_DIR=/Users/admin/Desktop/myapppdir/platforms/ios/build/sharedpch

Command finished with error code 2: /Users/admin/Desktop/myapppdir/platforms/ios/cordova/build --release,--device

ERROR building one of the platforms: Error: /Users/admin/Desktop/myapppdir/platforms/ios/cordova/build: Command failed with exit code 2
You may not have the required environment or OS to build this project

Error: /Users/admin/Desktop/myapppdir/platforms/ios/cordova/build: Command failed with exit code 2
    at ChildProcess.whenDone (/usr/local/lib/node_modules/phonegap/node_modules/cordova-lib/node_modules/cordova-common/src/superspawn.js:139:23)
    at emitTwo (events.js:87:13)
    at ChildProcess.emit (events.js:172:7)
    at maybeClose (internal/child_process.js:818:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)
```

**Comment by macdonst** [comment]
@yzargari try upgrading your project to use Cordova-iOS 4.0.1. Let me know if that doesn't work. Version 1.6.x of the plugin will make that a requirement.

Related to #456

**Comment by yzargari** [comment]
Hi Simon,

At your advice, I upgraded cordova ios to 4.0.1 but it didn't work initially, so I ended up creating a new project with cordova-ios 4.0.1, install all plugins copy the config.xml file and www directory.
It is now building fine (has some CFBundleIconFile referencing somewhere that wasn't existing, so I just removed the icon file attribute from the project to fix/workaround that).

Thanks a lot for the tip!
Joseph.
